server/responsewriter/gzip_test.go

```diff
@@ -35,53 +45,89 @@ func (d *DummyWriter) WriteHeader(int) {
-	assert.Equal(t, "gzip", gz.Header().Get("Content-Encoding"))
+	assert.Equal("", gz.Header().Get("Content-Length"))
+	assert.Equal("gzip", gz.Header().Get("Content-Encoding"))
 }
 
 // Gzip handler function should set content-type to "gzip" if accept-encoding header contains gzip
-func TestHandlerSetContent(t *testing.T) {
+func TestSetContentWithoutWrite(t *testing.T) {
+	assert := assert.New(t)
+
+	// Test content encoding header when write is not used
+	handlerFunc := Gzip(&DummyHandler{})
+
+	// Test when accept-encoding only contains gzip
 	rw := NewDummyWriter()
-	handler := &DummyHandler{}
-	req := &http.Request{}
+	req := NewRequest("gzip")
+	handlerFunc.ServeHTTP(rw, req)
+	// Content encoding should be empty since write has not been used
+	assert.Equal("", rw.Header().Get("Content-Encoding"))
 
-	req.Header = map[string][]string{}
-	handlerFunc := Gzip(handler)
+	// Test when accept-encoding contains multiple options, including gzip
+	rw = NewDummyWriter()
+	req = NewRequest("json, xml, gzip")
+	handlerFunc.ServeHTTP(rw, req)
+	assert.Equal("", rw.Header().Get("Content-Encoding"))
 
-	// test when accept-encoding only contains gzip
-	req.Header.Set("Accept-Encoding", "gzip")
+	// Test when accept-encoding is empty
+	req = NewRequest("")
+	rw = NewDummyWriter()
 	handlerFunc.ServeHTTP(rw, req)
-	assert.Equal(t, "gzip", rw.Header().Get("Content-Encoding"))
+	assert.Equal("", rw.Header().Get("Content-Encoding"))
 
-	// test when accept-encoding contains multiple options, including gzip
-	req.Header.Set("Accept-Encoding", "json, xml, gzip")
+	// Test when accept-encoding is is not empty but does not include gzip
+	req = NewRequest("json, xml")
+	rw = NewDummyWriter()
 	handlerFunc.ServeHTTP(rw, req)
-	assert.Equal(t, "gzip", rw.Header().Get("Content-Encoding"))
+	assert.Equal("", rw.Header().Get("Content-Encoding"))
 }
 
-// Gzip handler function should not change content-type if accept encoding does not contain gzip
-func TestHandlerForNonGzip(t *testing.T) {
+// Gzip handler function should set content-type to "gzip" if accept-encoding header contains gzip
+func TestSetContentWithWrite(t *testing.T) {
+	assert := assert.New(t)
+
+	// Test content encoding header when write is used
+	handlerFunc := Gzip(&DummyHandlerWithWrite{})
+
+	// Test when accept-encoding only contains gzip
+	req := NewRequest("gzip")
 	rw := NewDummyWriter()
-	handler := &DummyHandler{}
-	req := &http.Request{}
+	handlerFunc.ServeHTTP(rw, req)
+	// Content encoding should be gzip since write has been used
+	assert.Equal("gzip", rw.Header().Get("Content-Encoding"))
 
-	req.Header = map[string][]string{}
-	handlerFunc := Gzip(handler)
+	// Test when accept-encoding contains multiple options, including gzip
+	req = NewRequest("json, xml, gzip")
+	rw = NewDummyWriter()
+	handlerFunc.ServeHTTP(rw, req)
+	// Content encoding should be gzip since write has been used
+	assert.Equal("gzip", rw.Header().Get("Content-Encoding"))
```

Comment:
`.Get` only gets you the first response for that key, `json` could be hiding in there, should also check length to ensure you only have what you want in the header.